### PR TITLE
New Away Mission: Caves 

### PR DESCRIPTION
**Pull request by MMMiracles**
Adds a new away mission to explore called 'Caves'. 

Its based around an underground mining camp that's been abandoned. All that's left is the rusting camp and the rails that used to haul ore. Recent faults has caused the place to crack and open up holes and streams of magma to flow throughout the caves. There will be multiple layers to explore with an actual story attached to it.

There is four total levels to explore, each level getting more dangerous. There are multiple spots on the first two levels where players will spawn in once entering the mission, all spots are 'safe' as far as not instantly dying, but you can be stranded without proper gear (prepare before going in u dumbo).

REDDY4REVIEW LETS GO

:cl: MMMiracles
add: Adds a brand new away mission for the gateway called 'Caves'.
add: The away mission has multiple levels to explore and could be quite dangerous, running in without proper preparation is unadvised.
/:cl:

### CODE REVIEW

**Comment by WJohn** [comment]
I'd really appreciate a picture so I don't have to download the map and review its quality. Maybe send me one on IRC or something, I won't leak it.

**Comment by MMMiracles** [comment]
Ill hop in real quick and grab a map-wide screenshot and send you it on IRC, then.

**Comment by Jalleo** [comment]
Can we have a pic of the direct area ya arrive at?

**Comment by RemieRichards** [comment]
@MMMiracles probably best to give holes ```mouse_opacity = 0```, this makes it so you can't click them

**Comment by MMMiracles** [comment]
i accidentally a .dme

fuck

File: code/game/turfs/simulated/floor/plating.dm [hunk 1]
```diff
@@ -173,4 +173,59 @@
...
+		O.z = tp_z
+	else if (istype(H, /mob/living))
+		var/mob/living/M = H
+		M.visible_message("[M] falls down \the [src]!", \
+			"<span class='warning'>You fall down \the [src]!</span>")
+		M.x = tp_x
+		M.y = tp_y
+		M.z = tp_z
+		M.Weaken(6)
+		M.emote("scream")
```
Comment (Aranclanos) [inline code/game/turfs/simulated/floor/plating.dm]: oh god what are you doing nigger

File: code/game/turfs/simulated/floor/plating.dm [hunk 2]
```diff
@@ -173,4 +173,59 @@
...
+/turf/simulated/floor/plating/hole/update_icon()
+	if(!..())
+		return 0
+	if(!broken && !burnt)
+		if(smooth)
+			smooth_icon(src)
+	else
+		make_plating()
+		if(smooth)
+			smooth_icon_neighbors(src)
```
Comment (Aranclanos) [inline code/game/turfs/simulated/floor/plating.dm]: looks copypasted

File: code/game/turfs/simulated/floor/plating.dm [hunk 3]
```diff
@@ -173,4 +173,59 @@
...
+	smooth = SMOOTH_TRUE
+	canSmoothWith = null
+	mouse_opacity = 0
+	var/tp_x = 0 //hole drop-point
+	var/tp_y = 0
+	var/tp_z = 0
+
+/turf/simulated/floor/plating/hole/New()
+	..()
+	spawn(1)
```
Comment (Aranclanos) [inline code/game/turfs/simulated/floor/plating.dm]: random spawn()

File: code/game/turfs/simulated/floor/plating.dm (hunk 1 above)
Comment (MMMiracles) [inline code/game/turfs/simulated/floor/plating.dm]: i had this part looked over in coderbus by 3 people in a piratepad and none of them said anything bad about it. im just a sub-par coder trying to make shit for my map fuck

**Comment by MMMiracles** [comment]
been doing shit and dealing with a dying PSU. it should be in a final state by the end of the week.

if you could remove the conflict tag too since i fixed that it'd be nice.

**Comment by MMMiracles** [comment]
yeah so i cant really finish because i was relying on kor's lavaturf to get finished.
this entire map was based on lava being lethal so uhhhhhhhhhhhhhhhhh

**Comment by Aranclanos** [comment]
you still have a lot of work to do, do not hold everything because you have to wait on one little thing from another guy, its a common mistake that ruins projects

**Comment by KorPhaeron** [comment]
We have new smoothing code for turfs all that copypaste isnt needed is it?

You just set the var and it does it automatically

**Comment by KorPhaeron** [comment]
Your hole code doesnt have any check for things like projectiles which would logically fly over either.

I can do the chasm turf as well if you want

**Comment by MMMiracles** [comment]
If you want to make something better than what I managed to hobble together then please, go ahead.

**Comment by MMMiracles** [comment]
did adding that line to the .dme make travis kill itself or

File: code/modules/mob/living/simple_animal/hostile/skeleton.dm [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	see_in_dark = 8
+	layer = MOB_LAYER - 0.1
+	var/remains = /obj/effect/decal/remains/human
+
+/mob/living/simple_animal/hostile/skeleton/death(gibbed)
+	..(gibbed)
+	if(remains)
+		new remains (src.loc)
+	visible_message("<span class='danger'>The skeleton falls apart!</span>")
+	qdel(src)
```
Comment (Aranclanos) [inline code/modules/mob/living/simple_animal/hostile/skeleton.dm]: isnt this done by the parent? the deleting and creation of 'gibs'

File: code/game/turfs/simulated/floor/plating.dm [hunk 5]
```diff
@@ -177,4 +177,44 @@
...
+	else if (istype(H, /mob/living))
+		var/mob/living/M = H
+		M.visible_message("[M] falls down \the [src]!", \
+			"<span class='warning'>You fall down \the [src]!</span>")
+		M.x = tp_x
+		M.y = tp_y
+		M.z = tp_z
+		M.Weaken(6)
+		M.emote("scream")
+		M.adjustBruteLoss(20)
```
Comment (Aranclanos) [inline code/game/turfs/simulated/floor/plating.dm]: this teleportation method is still awful, please use old systems and do not create a new one

File: code/modules/mob/living/simple_animal/hostile/skeleton.dm (hunk 4 above)
Comment (MMMiracles) [inline code/modules/mob/living/simple_animal/hostile/skeleton.dm]: not the way i wanted it done, as far as im aware of.

**Comment by MMMiracles** [comment]
travis don't do this to me please

**Comment by KorPhaeron** [comment]
@MMMiracles 

It's kill

**Comment by MMMiracles** [comment]
it's something about the change ops folder but I don't know what psfffff

**Comment by Jordie0608** [comment]
Restarted build, was an issue with master repo that's fixed now.

**Comment by MMMiracles** [comment]
the dream lives on

**Comment by Jordie0608** [comment]
I fixed it for you but: double quotes in changelogs will break the post-merge changelog compile, use single quotes.

**Comment by KorPhaeron** [comment]
The holes warp you to space

There is no ladder from area 2 to area 3

**Comment by KorPhaeron** [comment]
The holes also need a sprite they'er basically invisible

**Comment by MMMiracles** [comment]
@KorPhaeron 

If people are getting warped into space that probably means I need to move to a different method of moving for the hole unless leaving the 'z' coordinate blank actually accounts for your current z and doesn't just warp you into nullspace.

You kinda have to drop down into the 2nd level's hole to reach the 3rd level, which probably isn't the best design but its suppose to be like a 'once you go down you can't go back' ordeal.

The holes do a sprite its just that they're really shitty and could probably be better.

**Comment by MMMiracles** [comment]
iamreddy.

**Comment by Boredone** [comment]
Minor complaint about the last zone based on the round where it got tested. Automatic Mob Spawners+Long delay before anyone can even enter the away mission, let alone reach the end=TONS OF SKELETONS.

**Comment by MrStonedOne** [comment]
HELP: Dr. Aura/(Ebony-And-Ivory) (?) (PP) (VV) (SM) (FLW) (TP) (REJT): TIL the Ridley can drill the ladder into non-existence

**Comment by Nitrousoxide** [comment]
So apparently you can delete the ladders between levels with a ripley and trap people forever on levels.

**Comment by MrStonedOne** [comment]
DEAD: ADMIN(MrStonedOne) says, "just make a subtype of the ladder, and have it return QDEL_HINT_LETMELIVE on it's destroy proc"

**Comment by MrStonedOne** [comment]
HELP: Smittn/(Terry Mcfall) (?) (PP) (VV) (SM) (FLW) (TP) (REJT): found a bug, drilling the basalt floors hields infinite metal

**Comment by KorPhaeron** [comment]
Can you dig your way out of the different floors?

**Comment by MrStonedOne** [comment]
no, but that would be cool.

**Comment by KorPhaeron** [comment]
@MrStonedOne 

I mean breaking out into space through the walls

**Comment by MrStonedOne** [comment]
no, they are /turf/indestructible/rock

**Comment by KorPhaeron** [comment]
@MMMiracles 

The skeletons attack corpses forever.

Set their stat_attack to 1, not 2

**Comment by Zilenan91** [comment]
There needs to be plentiful amounts of mesons scattered around, because you can wander around and have no idea where you're going.

**Comment by Zilenan91** [comment]
Make there be no delay to this linking. The whole point of the delay is to lock out good equipment from the start of the round. This away mission has no good equipment as far as I know

**Comment by MMMiracles** [comment]
Feedback has been addressed and issues were fixed.

Basalt floors shouldn't be infinite metal anymore and ladders can't be drilled out of existence anymore via ripley.

Spawn rates of spawners have been nerfed to more reasonable levels to make exploring less of a death sentence and to spare the server CPU.

More loot has been scattered about for those interested in more than ore. Might even be helpful in dealing with all those mining monsters..

The first level has been equipped with a mini-medbay area for those who were foolish enough to not bring supplies beforehand. Obtaining the actually useful kits might be a little difficult, though.

A few hardsuits have been placed on the map on the first level, but will probably be a little difficult to obtain.

Some pathways across the lava have been widened due to people complaining about accidentally walking into the lava. I say people should be more careful near giant lakes of lava but that's just me.

**Comment by MMMiracles** [comment]
Local testing on my end shows everything working properly and I've made sure every level is reachable on a normal means. The loot mix should be a decent balance between risk and reward so I'm finally done editing.

**Comment by Qbopper** [comment]
Just finished an away mission on this - fuck, this map is awesome, you did a fantastic job

The medbot at the one ladder room was a super smart idea - I loved how we had to wrok as a team to survive and make it through. If it wasn't for the captain that round I'dve died on the first floor

Ignore the morons whining about lava, the one tile paths worked fine, you just need to not be a moron and pay attention.

I think that using the frag nades makes the lava turn into regular plating - not sure what was up with that, maybe I imagined it?

The tweaks to skeleton spawns seem to have worked out well - between the laser guns, SORDs, and accelerators, the 4/5 of us did alright against the skeletons - enough that we had to be on our toes, but not so many that it was a fustercluck of mobs.

All in all, absolutely fantastic time for my first away mission and I hope to do it again some time, and I hope you make more

**Comment by KorPhaeron** [comment]
Looting the dead for spare internals was pretty fun. Overall great stuff. One of the best missions by far

My one complaint is that a spicy burritto is a bit anti climactic. Hell there isn't really a finale in general.

Space battle you've repelled a massive syndicate assault and you get that ridiculous gun, wild west you reach the wish granter and either perpetuate or end a terrible curse, etc.

Would be nice to have something that feels like an overall goal tied to the caves story. A boss creature or something else special. I can help you code something if you'd like.

**Comment by MMMiracles** [comment]
If you're willing to help with a boss, I'd love to see some sort of skeleton necromancer that makes nearby remains rise into living skeletons again while shooting his own dark magic shit.

Something like special forcewalls that only vanish after the mob's death alone would be a great start for a 'boss' mob.

**Comment by duncathan** [comment]
If lavaworld never happens, this mission doesn't make for a terrible substitute.

**Comment by KorPhaeron** [comment]
@duncathan

Inching towards completion I swear

**Comment by Anonus** [comment]
Also if you're using cult turfs I suggest using the cult subtype of forcewalls instead of normal forcewalls because they look a lot better on cult stuff.

**Comment by KorPhaeron** [comment]
We have cult forcewalls?

**Comment by Zilenan91** [comment]
Yeah someone sprited them not too long ago.

Skeletal necromancer who hurls fireballs and summons skeletons? Can I contain my hype?

**Comment by MMMiracles** [comment]
I've done more feedback changes after the few rounds I observed with both watching people play through and complaints.

Apparently lava isn't suppose to be really deadly when you walk directly into it according to a few people but what do I know.

**Comment by KorPhaeron** [comment]
You better not have changed the narrow lava paths.

They're really fun to walk down

**Comment by MMMiracles** [comment]
I think I changed a total of two narrow paths, and that was because I added in some structures to help move mechs across the cave levels.

This has had two separate playtests on the server over multiple rounds with feedback changes, I think you can remove the testing tag now.

**Comment by MMMiracles** [comment]
merge y/y

**Comment by RemieRichards** [comment]
@MMMiracles was actually about to ask if it was ready lol.

**Comment by MMMiracles** [comment]
17 days later and its finally over.

**Pull request merged**